src/npcmove.cpp

```diff
@@ -1105,8 +1105,8 @@ void npc::move_to(int x, int y)
             moves -= run_cost(g->m.combined_movecost(posx(), posy(), x, y), diag);
             if (g->m.tr_at(x, y) != tr_null) { // NPC stepped on a trap!
                 trap *tr = traplist[g->m.tr_at(x, y)];
-                if (!this->avoid_trap(tr, x, y)) {
-                    tr->trigger(this, x, y);
+                if (!this->avoid_trap( tripoint( x, y, posz() ), tr )) {
+                    tr->trigger( tripoint( x, y, posz() ), this );
```

Comment:
Same here: just a few line above the position of the NPC has been set to (x,y), so you can use `pos3()` here instead of creating the tripoint manually.